### ISSUE TITLE
ENH: Optuna Integration

### ISSUE THREAD

**Issue by lithomas1** [lithomas1/sherlock#38]
Use optuna to help tune model hyperparameters, optimizers, learning rate, etc.
